src/firefly/js/ui/ColorPicker.jsx

```diff
@@ -24,25 +39,41 @@ export function showColorPickerDialog(color, callbackOnOKOnly, callbackOnBoth, c
 
 var lastEv;
 
-function ColorPickerWrapper ({callback,color,callbackOnOKOnly, callbackOnBoth}) {
+function ColorPickerWrapper ({callback,color,callbackOnOKOnly, callbackOnBoth,
+                                 presetAlpha,  presetColors=DEF_PRESET_COLORS}) {
     const [currentColor, setCurrentColor] = useState(color);
 
     const updateColor= (ev) => {
+        lastEv=ev;
         const {r,g,b,a}= ev.rgb;
         setCurrentColor(`rgba(${r},${g},${b},${a})`);
-        callback(ev,false);
+        if (!callbackOnOKOnly) callback?.(ev,false);
     };
 
+    const psColors= !presetAlpha ? presetColors :
+        presetColors.map( (c) => {
+            const [r,g,b]= getRGBA(c);
+            return `rgba(${r},${g},${b},${presetAlpha})`;
+        });
+
+
     return (
         <div>
-            <SketchPicker
-                         onChangeComplete={ (ev) => {
-                             lastEv=ev;
-                             if (!callbackOnOKOnly) updateColor(ev);
-                         } }
-                         color={currentColor} />
-            <CompleteButton onSuccess={() => callbackOnOKOnly||callbackOnBoth ? callback(lastEv,true): null}
-                            dialogId='ColorPickerDialog'/>
+            <div title={dialogTip}>
+                <SketchPicker color={currentColor} presetColors={psColors} onChangeComplete={updateColor} />
+            </div>
```

Comment:
I think that is purposeful. Because the image overlay use the option to update immediately (`callbackonBoth`). Did you do that in your PR? Did you test it with the image catalog overlays? I think I am keeping the same functionality as before. I don’t think large image catalogs could keep up with the sort of updating.

Reply:
I thought about this overnight. We could you `onChange` for UI updates only. Then onChangeComplete could do what it is now, without breaking existing functionality.

Reply:
Exactly.  Also, instead of a callback and 2 flags, you could allow a callback for each event `onOk` and `onChange`.  onChange is triggered by onChangeComplete.  This would make it clear.

Reply:
I agree but with the flags, but in an effort to minimize changes for this release, I will change the flags in the future. The drawing layer code is complex enough the I would like to hold off on that until the next release.